### PR TITLE
[MU4] vst_scanning

### PR DESCRIPTION
**Pull request by vpereverzev**

### CODE REVIEW

File: src/framework/vst/internal/vstmodulesmetaregister.cpp [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+}
+
+bool VstModulesMetaRegister::isEmpty() const
+{
+    return m_metaMap.empty() && m_paths.empty();
+}
+
+void VstModulesMetaRegister::readMetaList()
+{
+    m_file.open(io::IODevice::OpenMode::ReadWrite);
```
Comment (RomanPudashkin) [inline src/framework/vst/internal/vstmodulesmetaregister.cpp:138]: IODevice::OpenMode::ReadOnly would be better here

File: src/framework/vst/internal/vstconfiguration.h [hunk 2]
```diff
@@ -23,17 +23,22 @@
...
 class VstConfiguration : public IVstConfiguration
 {
+    INJECT(vst, framework::IGlobalConfiguration, globalConfig)
 public:
     void init();
 
     io::paths_t userVstDirectories() const override;
     void setUserVstDirectories(const io::paths_t& paths) override;
     async::Channel<io::paths_t> userVstDirectoriesChanged() const override;
+    io::path_t knownPluginsList() const override;
```
Comment (RomanPudashkin) [inline src/framework/vst/internal/vstconfiguration.h:41]: What do you think about knownPluginsFile?

File: src/framework/vst/internal/vstmodulesmetaregister.cpp (hunk 1 above)
Comment (RomanPudashkin) [reply src/framework/vst/internal/vstmodulesmetaregister.cpp:138]: Hmm, now I understand why it's necessary

**Review by RomanPudashkin — APPROVED** [review]

File: src/framework/vst/internal/vstconfiguration.h (hunk 2 above)
Comment (vpereverzev) [reply src/framework/vst/internal/vstconfiguration.h]: actually, forgot to rename it

**Pull request merged**